lib/Driver/UnixToolChains.cpp

```diff
@@ -384,9 +387,12 @@ toolchains::GenericUnix::constructInvocation(const StaticLinkJobAction &job,
 
   ArgStringList Arguments;
 
+  const char *AR;
   // Configure the toolchain.
-  const char *AR =
-      context.OI.LTOVariant != OutputInfo::LTOKind::None ? "llvm-ar" : "ar";
+  if (getTriple().isAndroid())
+    AR = "llvm-ar";
+  else
+    AR = context.OI.LTOVariant != OutputInfo::LTOKind::None ? "llvm-ar" : "ar";
```

Comment:
It is still my opinion that if we are going to keep adding `isAndroid()` branches to the `GenericUnix` toolchain, it would be better to add a `Android` toolchain (like OpenBSD and Cygwin already have) and stop adding special branches for Android. Not something to hold the PR, but something you should think about if these special cases keep popping up.

Reply:
Sure, I'll look into spinning that out.